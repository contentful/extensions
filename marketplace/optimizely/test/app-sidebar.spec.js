import React from 'react';
import { render } from '@testing-library/react';

import AppSidebar from '../src/app-sidebar';

describe('AppSidebar', () => {
  function makeSdk(hasExperiment = false) {
    return {
      window: {
        startAutoResizer: () => {},
        stopAutoResizer: () => {}
      },
      entry: {
        fields: {
          experimentId: {
            onValueChanged: jest.fn(),
            getValue: jest.fn(() => (hasExperiment ? 'test-experiment' : undefined))
          }
        }
      }
    };
  }

  const mockClient = {
    getExperimentUrl: jest.fn(id => `experiment-url-${id}`),
    getAllExperimentsUrl: jest.fn(() => 'all-experiment-url')
  };
  it('should match snapshot for disabled', () => {
    const { container } = render(<AppSidebar sdk={makeSdk()} client={mockClient} />);

    expect(container).toMatchSnapshot();
  });
  it('should match snapshot for enabled', () => {
    const { container } = render(<AppSidebar sdk={makeSdk(true)} client={mockClient} />);

    expect(container).toMatchSnapshot();
  });
});
