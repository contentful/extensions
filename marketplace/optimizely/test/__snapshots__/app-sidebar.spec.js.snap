// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSidebar should match snapshot for disabled 1`] = `
<div>
  <div>
    <a
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-82qlwu Button__Button--primary___JImeO Button__Button--disabled___1E20M Button__Button--full-width___3Fmpo"
      data-test-id="cf-ui-button"
      disabled=""
      target="_blank"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          View in Optimizely
        </span>
      </span>
    </a>
    <a
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-82qlwu Button__Button--muted___2Wair Button__Button--full-width___3Fmpo"
      data-test-id="cf-ui-button"
      href="all-experiment-url"
      target="_blank"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          View all experiments
        </span>
      </span>
    </a>
  </div>
</div>
`;

exports[`AppSidebar should match snapshot for enabled 1`] = `
<div>
  <div>
    <a
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-82qlwu Button__Button--primary___JImeO Button__Button--full-width___3Fmpo"
      data-test-id="cf-ui-button"
      href="experiment-url-test-experiment"
      target="_blank"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          View in Optimizely
        </span>
      </span>
    </a>
    <a
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-82qlwu Button__Button--muted___2Wair Button__Button--full-width___3Fmpo"
      data-test-id="cf-ui-button"
      href="all-experiment-url"
      target="_blank"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          View all experiments
        </span>
      </span>
    </a>
  </div>
</div>
`;
